### DISCUSSION TITLE
Seeking Guidance for Akismet Addon + form submissions Migration in Statamic v2 to v4 Upgrade

### DISCUSSION THREAD

**Discussion by YoBigRube** [silentzco/statamic-akismet#51] [Q&A]
I just purchased the Akismet addon and am seeking guidance for the final stage of a Statamic migration from v2 (Akismet v1.8) to v4. I haven't yet completed step [#8](https://github.com/statamic/migrator) with the Statamic migrator in case it's needed for this part of the process. Initially, I had issues migrating from v2 to v4 due to installed addons, so I decided to only migrate Akismet and purchased the updated version.

To protect privacy, form submissions reside only on the production server. I haven't pulled them down locally, and they're not tracked in the git repository. During the content migration, I excluded contact forms. The plan is to manually download all submissions from the production server and migrate them into the v4 project local environment just before finishing the migration.

Current Setup

 • Form submissions: `"site/storage/forms"` (YAML files)
 • Akismet: `"site/storage/addons/Akismet"` (PHP files)
Questions

 1. File Locations: In my v4 project, where should I store the form submissions from v2 (.yaml and .php) to Composer install the Akismet addon and run the spam:convert+add:extension commands as per the install instructions?

 2. Migration of Submissions: In the control panel, there are fewer than 25 submissions in "cp/Forms" but about 2.6k submissions in "cp/Spam Queue." How can I properly migrate these to ensure the client retains their spam training and form submissions?

Thanks in advance,

**Comment by edalzell** [comment]
1. you'll need to install the current version, you can't "move" the existing one. The form submissions should move across with the statamic update, if not, they go in `/storage/statamic/forms` I think
2. why do they need them? They should handle those then start fresh imo

**Reply by YoBigRube** [reply]
I wanted to move the Akismet spam learning data; I was hoping they'd benefit from the 2k submissions in the spam queue - if that's possible. Unless I'm misunderstanding how that all works.

**Reply by edalzell** [reply]
that's not how the learning works. The learning only happens when you mark something as spam. Having it be in the queue is just for you in case it's a false positive.

So if you had already moved them manually to the queue in v2, the learning is already done.
